spynnaker/pyNN/models/neuron/synapse_io.py

```diff
@@ -316,65 +322,111 @@ def get_synapses(
                     delayed_connections["source"] - pre_vertex_slice.lo_atom)
 
             # Get the data
-            max_delayed_row_length, delayed_row_data = \
-                self._get_max_row_length_and_row_data(
-                    delayed_connections, delayed_row_indices,
-                    pre_vertex_slice.n_atoms * n_delay_stages,
-                    post_vertex_slice, n_synapse_types, population_table,
-                    synapse_info.synapse_dynamics)
+            delayed_row_data = self._get_row_data(
+                delayed_connections, delayed_row_indices,
+                pre_vertex_slice.n_atoms * n_delay_stages, post_vertex_slice,
+                n_synapse_types, synapse_info.synapse_dynamics,
+                max_row_info.delayed_max_n_synapses,
+                max_row_info.delayed_max_words)
             del delayed_row_indices
         del delayed_connections
 
-        return (row_data, max_row_length, delayed_row_data,
-                max_delayed_row_length, delayed_source_ids, stages)
+        return (row_data, delayed_row_data, delayed_source_ids, stages)
 
-    def read_synapses(
+    @staticmethod
+    def _rescale_connections(
+            connections, machine_time_step, weight_scales, synapse_info):
+        # Return the delays values to milliseconds
+        connections["delay"] /= 1000.0 / machine_time_step
+        # Undo the weight scaling
+        connections["weight"] /= weight_scales[synapse_info.synapse_type]
+        return connections
+
+    def read_undelayed_synapses(
```

Comment:
the code for read_delayed_synapses and read_undelayed_syanspes are clones if you throw static and plastic calls.

Reply:
given its called by read_synapses. you could put those methods into the calls tehre and thus have only 1 function

Reply:
Done this better now